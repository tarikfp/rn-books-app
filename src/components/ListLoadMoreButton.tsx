import * as React from 'react';
import { BaseButton } from 'react-native-gesture-handler';
import AntDesign from 'react-native-vector-icons/AntDesign';
import { useAppTheme } from '~theme/core';

type Props = {
  onPress: () => void;
  testID?: string;
};

export const ListLoadMoreButton: React.FC<Props> = ({ onPress, testID }) => {
  const { colors } = useAppTheme();
  return (
    <BaseButton testID={testID} onPress={onPress}>
      <AntDesign disabled name="pluscircleo" size={35} color={colors.primary} />
    </BaseButton>
  );
};
