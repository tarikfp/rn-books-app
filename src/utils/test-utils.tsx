import { ThemeProvider } from '@shopify/restyle';
import { render, RenderOptions } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { theme } from '~theme/core';
import { ThemePreferenceContext } from '~theme/useThemePreference';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ThemePreferenceContext.Provider
      value={{
        isDarkTheme: false,
        toggleTheme: () => {},
      }}
    >
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </ThemePreferenceContext.Provider>
  );
};

const withAllTheProviders = (ui: React.ReactNode) => (
  <AllTheProviders children={ui} />
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderer = (ui: React.ReactElement) =>
  renderer.create(withAllTheProviders(ui));

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { customRender as render };
// override renderer method
export { customRenderer as createRenderer };
