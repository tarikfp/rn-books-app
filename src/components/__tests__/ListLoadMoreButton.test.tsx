import { fireEvent } from '@testing-library/react-native';
import React from 'react';
import { ListLoadMoreButton } from '~components/ListLoadMoreButton';
import { render } from '~utils/test-utils';

describe('ListLoadMoreButton Component', () => {
  it('matches snapshot', () => {
    const onPressMock = jest.fn();
    const { toJSON } = render(<ListLoadMoreButton onPress={onPressMock} />);

    expect(toJSON()).toMatchSnapshot();
  });
  it('triggers onPress mock when clicked', () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(
      <ListLoadMoreButton
        testID="list-load-more-button"
        onPress={onPressMock}
      />,
    );

    const button = getByTestId('list-load-more-button');
    fireEvent.press(button);

    expect(onPressMock).toHaveBeenCalled();
  });
});
