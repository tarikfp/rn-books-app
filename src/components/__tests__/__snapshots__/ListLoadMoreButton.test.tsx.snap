// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListLoadMoreButton Component matches snapshot 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  onPress={[MockFunction]}
>
  <Text
    allowFontScaling={false}
    disabled={true}
    selectable={false}
    style={
      [
        {
          "color": "#3EBCB9",
          "fontSize": 35,
        },
        undefined,
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    î˜¥
  </Text>
</RNGestureHandlerButton>
`;
